contracts/Company/Extras/ExitController.sol

```diff
@@ -0,0 +1,277 @@
+        uint32 newState,
+        uint32 timestamp
+    );
+
+    event LogProceedsPayed(
+        address investor,
+        uint256 amountEquityTokens,
+        uint256 amountPayed
+    );
+
+    event LogProceedsManuallyResolved(
+        address lostAddress,
+        address newAddress,
+        uint256 amountEquityTokens,
+        uint256 amountPayed
+    );
+
+    ////////////////////////
+    // Types
+    ////////////////////////
+
+    // defines state machine of the exit controller
+    enum State {
+        Setup, // Initial state
+        Payout, // Users can claim eur-t for tokens
+        ManualPayoutResolution // Nominee can manually resolve payouts, user initiated payout is disabled
+    }
+
+    ////////////////////////
+    // Immutable state
+    ////////////////////////
+
+    // a root of trust contract
+    Universe private UNIVERSE;
+    IERC223Token private EURO_TOKEN;
+    // equity token from ETO
+    IEquityToken private EQUITY_TOKEN;
+
+    ////////////////////////
+    // Mutable state
+    ////////////////////////
+
+    // controller lifecycle state
+    State private _state;
+
+    // exit values get set when exit proceedings start
+    uint256 private _exitEquityTokenSupply = 0;
+    uint256 private _exitAquisitionPriceEurUlps = 0;
+    uint256 private _manualPayoutResolutionStart = 0;
+
+    // keep record of manually resolved payout
+    mapping(address => bool) private payoutManuallyResolved;
+
+    ////////////////////////
+    // Modifiers
+    ////////////////////////
+
+    ////////////////////////
+    // Constructor
+    ////////////////////////
+
+    constructor(
+        Universe universe,
+        IEquityToken equityToken
+    )
+        public
+        Agreement(universe.accessPolicy(), universe.forkArbiter())
+        Reclaimable()
+    {
+        UNIVERSE = universe;
+        EURO_TOKEN = UNIVERSE.euroToken();
+        EQUITY_TOKEN = equityToken;
+        _state = State.Setup;
+    }
+
+    //
+    // Implements IControllerGovernance
+    //
+    function state()
+        public
+        constant
+        returns (State)
+    {
+        return _state;
+    }
+
+    function payoutInfo()
+        public
+        constant
+        returns (uint256 exitEquityTokenSupply, uint256 exitAquisitionPriceEurUlps, uint256 manualPayoutResolutionStart)
+    {
+        return (
+            _exitEquityTokenSupply, _exitAquisitionPriceEurUlps, _manualPayoutResolutionStart
+        );
+    }
+
+    // calculate how many eurotokens one would receive for the given amount of tokens
+    function eligibleProceedsForTokens(uint256 amountTokens)
+        public
+        constant
+        returns (uint256)
+    {
+        if (_state == State.Setup ) {
+            return 0;
+        }
+        // calculate the amount of eligible proceeds based on the total equity token supply and the
+        // acquisition price
+        return Math.mul(_exitAquisitionPriceEurUlps, amountTokens) / _exitEquityTokenSupply;
```

Comment:
Sadly I don't know about high precision math but there is also `Math.divRound` maybe it would be better her than simple `'`

Reply:
The div round does a round up from .5 upwards, and I thought it would be safer to always round down the way / does. Or what do you think?